src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/Search/SortValueJObjectGenerator.cs

```diff
@@ -0,0 +1,119 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using EnsureThat;
+using Microsoft.Health.Fhir.Core.Features.Search.SearchValues;
+using Microsoft.Health.Fhir.CosmosDb.Features.Search;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Fhir.CosmosDb.Features.Storage.Search
+{
+    public class SortValueJObjectGenerator : ISearchValueVisitor
+    {
+        private string _prefix;
+
+        private JObject CurrentEntry { get; set; }
+
+        public JObject Generate(SortValue entry)
```

Comment:
I know this is how it's done today, but one day, we'll need to reimplement this this using Utf8JsonWriter/Utf8JsonWriter instead of allocating all these intermediate objects.